lib/chef/provider/service/redhat.rb

```diff
@@ -60,9 +60,15 @@ def define_resource_requirements
             a.failure_message Chef::Exceptions::Service, "#{chkconfig_file} dbleoes not exist!"
           end
 
-          requirements.assert(:start, :enable, :reload, :restart) do |a|
+          requirements.assert(:enable) do |a|
+            a.assertion { !@service_missing }
+            a.failure_message Chef::Exceptions::Service, "#{new_resource}: Service is not known to chkconfig."
+            a.whyrun "Assuming service would be disabled. The init script is not presently installed."
```

Comment:
minor text nit that this should be "Assuming service would be *enabled*" rather than "disabled"